sr5824-msr9732-sm9453/new_project.md

```diff
@@ -0,0 +1,148 @@
+streaming data and benchmarked with real time applications[5].
+
+We want to analyse and benchmark unsupervised learning on the cluster data[6] to 
+evaluate the prospects of AIOps in real world usage.
+
+# Proposed Solution
+
+To get the most value out of the AIOps system, it should be deployed as an independent platform that 
+ingests data from all IT monitoring sources, and acts as a central system of engagement.
+
+Such a platform must be powered by three major types of algorithms that fully automate and 
+streamline three key dimensions of IT operations monitoring:
+
+1. Detection - Machine Learning models trained on past dataset, which would detect any aberration in the
+               metrics and would raise a flag. This would be the entry point of the system.
+   
+2. Triaging - Identifying root causes of problems and recurring issues, so that you can take action 
+              on what has been discovered.
+   
+3. Remediate - Take an appropriate action to avoid/tackle the problem at hand.
+
+These steps occur in a sequential manner, i.e, Once we detect an incident, we triage it and then remediate it.
+
+As part of this project, we are going to deploy the AIOps lifecycle. The way we plan to do this is
+as follows:
+
+1. Detection:
+    We will have a detection API (Flask application) being served from CloudLab. We will have pre-trained 
+    models that would sit on the local file system (we would just pickle them and store them. We are not 
+    making it more complex as our main goal is not to have the complete system. Our main goal is to evaluate
+    all the different techniques).
+   
+2. Triage:
+    This is the most difficult part to accomplish. We do not plan to come up with a complete RCA. We are 
+    going to mock this part with rules, to showcase how all the different parts are stitched together. We
+    are going to use StackStorm[7] for this. StackStorm would have sensors, which would poll the data, call 
+    our detection API, based on the Anomaly Scores, match a corresponding rule and take an appropriate action.
+   
+3. Remediate:
+    This part contains the set of actions that we would want to take. These actions can be as simple as
+    restarting a container. This would again be part of the StackStorm ecosystems. Actions are scripts
+    which would be called by StackStorm when a particular rule is matched against it.
+   
+The reason we want to come-up with the entire AIOps Cycle is to showcase how different detecting algorithms
+can have adverse effects in real world. For example: if we miss an incident, there would not be any rules
+matched and thus there would not be any action taken on it.
+
+The main part of our project is going to be evaluating different Unsupervised Anomaly Detection Algorithms. 
+
+We plan to serve the following models using the API:
+
+1. Statistical:\
+    a. Autoregression (AR)\
+    b. Moving Average (MA)\
+    c. ARIMA\
+    d. SARIMA\
+    e. SARIMAX
```

Comment:
What features do you plan to use for all the AR based algorithms?

Reply:
We plan to approach these algorithms using a univariate approach, and the feature we plan on using is CPU Utilization (we can also use other metrics and then study the correlation to understand how each one is affecting the other). Essentially, at any given training our feature vector would only hold a single value (which to start off with would be CPU Utilization). The dataset we are going to use is the one published by Alibaba (https://github.com/alibaba/clusterdata/blob/master/cluster-trace-v2018/schema.txt).

Reply:
All of these methods do support multivariate features too (https://www.itl.nist.gov/div898/handbook/pmc/section4/pmc45.htm). While Alibaba only gives you memory and CPU, I would consider using both since depending on the job CPU might be insufficient.

Reply:
Yes. They do work on multi-variate. However, I am unsure of how well would these do, because we do not have any established relationship between different metrics yet. But it would be nice to check it out. We will try to have a multivariate approach too and add it to our analysis.